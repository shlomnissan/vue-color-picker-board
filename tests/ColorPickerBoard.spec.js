import { shallowMount } from '@vue/test-utils';
import ColorPickerBoard from '../src/ColorPickerBoard.vue';

describe('ColorPickerBoard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ColorPickerBoard);
  });

  it("is vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  describe(":props", () => {
    it(":width", () => {
      wrapper.setProps({ width: 100 });
      expect(wrapper.find("#canvas").attributes('width')).toEqual("100");
    });

    it(":height", () => {
      wrapper.setProps({ height: 100 });
      expect(wrapper.find("#canvas").attributes('height')).toEqual("100");
    });

    it(":defaultColor", () => {
      const wrapper = shallowMount(ColorPickerBoard, {
        propsData: {defaultColor: "#FF00AA"}
      });
      expect(wrapper.find("#canvas").element.style.backgroundColor).toEqual("rgb(255, 0, 170)");
    });
  });

  describe("@events", () => {
    it("@onSelection", () => {
      wrapper.find('#canvas').trigger('click');
      expect(wrapper.emitted()["onSelection"]).toBeTruthy();
    });
  });
});