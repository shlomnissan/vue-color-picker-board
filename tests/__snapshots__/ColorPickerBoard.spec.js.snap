// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerBoard renders correctly 1`] = `
<div
  class="component color-picker"
>
  <canvas
    height="100"
    id="canvas"
    style="background-color: rgb(75, 186, 221);"
    width="800"
  />
</div>
`;
